src/libraries/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Hosting/AssemblyCatalog.cs

```diff
@@ -543,6 +543,8 @@ private void ThrowIfDisposed()
         private string GetDisplayName() =>
             $"{GetType().Name} (Assembly=\"{Assembly.FullName}\")";   // NOLOC
 
+        [UnconditionalSuppressMessage("Single file", "IL3000: Avoid accessing Assembly file path when publishing as a single file",
+            Justification = "Setting a CodeBase is single file compatible")]
```

Comment:
Did you look at only making the getter for CodeBase "unsafe" and leave the setter without an attribute?

Reply:
Yes, the annotation is in the getter of codebase in the file `src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyName.cs` in this PR.
I didn't change the analyzer behavior for now since it will mean more work to special case the codebase getter in the analyzer. The plan is to annotate first, then get rid of the special casing in the analyzers for CodeBase/EscapedCodeBase since they are deprecated and provide a misleading warning, and once the analyzer doesn't check for those we can delete the UnconditionalSuppressMessage and tests in the linker. Also cannot remove it for now since the annotations are not in place and we wouldn't warn.